ecg_data.py

```diff
@@ -1,12 +1,100 @@
-class hrm:
+import numpy
+import pandas
+
+
+class ecg_data:
     'This is a class'
-    def __init__(self, filename, avgT1,avgT2,bradyT,bradyThresh,tachyT,tachyThresh):
-        self.avgT1 = 0.0
-        self.avgT2 = 10.0
 
-    #def instantHr(self):
+    def __init__(
+            self,
+            filename,
+            begin_time=0,
+            end_time=10,
+            bradyT=5,
+            bradyThresh=60,
+            tachyT=5,
+            tachyThresh=100):
+
+        # put peak detection here
+        data = pandas.read_csv(
+            filename, converters={
+                "times": float, "voltages": float})
+        voltages = data.voltages.values
+        finalTimes = [0]
+        avgvoltage = numpy.average(voltages)
+
+        threshvoltage = avgvoltage * 2
+
+        peaks = numpy.where(voltages >= threshvoltage)
+        peaks1 = peaks[0]
+        peaks2 = peaks1.tolist()
+        boxcar = []
+        for i in range(1, len(peaks2) - 1):
+            if (voltages[peaks2[i]] >= voltages[peaks2[i - 1]]) and \
+                    (voltages[peaks2[i]] >= voltages[peaks2[i + 1]]):
+                recentval = finalTimes[len(finalTimes) - 1]
+                finalTimes.append(peaks2[i])
+                if (peaks2[i] - recentval <= 50):
+                    finalTimes.pop()
+        finalTimes.pop(0)
+
+        self.times = finalTimes
+        self.inst = self.instantHr(begin_time)
+        self.avg = self.averageHR(begin_time, end_time)
+        self.ano = self.anomalyHr(bradyT, bradyThresh, tachyT, tachyThresh)
+
+    def instantHr(self, target_time = 0):
```

Comment:
why is `target_time` not an attribute set in `__init__()`?

Reply:
We wanted the people to be able to update the target time, so wont it need to be an input? Instead of having them update the attribute and run the method again? Or is it better practice for them to do that?

Reply:
You could use a "setter" method... which would involve updating an attribute value and then automatically triggering all of the dependent methods to be re-run and updated.  Check out: https://www.python-course.eu/python3_properties.php, which highlights the use of the `@property` decorator.

Reply:
Is this new update what you were talking about?